libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFAuthenticationManager.m

```diff
@@ -471,7 +450,7 @@ - (void)logoutUser:(SFUserAccount *)user
         [self revokeRefreshToken:user];
         return;
     }
-    
+    user.userLoggingOut = YES;
```

Comment:
This should really happen at [Line 430](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/1939/files#diff-4aedc82766c844e15a84507c44206a64R430), and ideally all be wrapped in an atomic block, to ensure that the meat of this method is only executed once.